### PR TITLE
Tree router continue

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

File: test/Microsoft.AspNet.Routing.Tests/Tree/TreeRouterTest.cs [hunk 1]
```diff
@@ -581,8 +585,10 @@ public void AttributeRoute_GenerateLink_RespectsPrecedence(string firstTemplate,
         [InlineData("template/{parameter:int:range(1,20)?}", "/template", null)]
         [InlineData("template/{parameter:int:range(1,20)?}", "/template/5", 5)]
         [InlineData("template/{parameter:int:range(1,20)?}", null, 21)]
-        public void AttributeRoute_GenerateLink_OptionalInlineParameter
-            (string template, string expectedPath, object parameter)
+        public void TreeRouter_GenerateLink_OptionalInlineParameter(
+            string template,
+            string expectedPath,
+            object parameter)
```
Comment (dougbu) [inline test/Microsoft.AspNet.Routing.Tests/Tree/TreeRouterTest.cs]: Doesn't this just fit on one line?

File: test/Microsoft.AspNet.Routing.Tests/Tree/TreeRouterTest.cs [hunk 2]
```diff
@@ -202,8 +203,10 @@ public class TreeRouterTest
         [InlineData("template/{parameter:int?}", "/template/5", true)]
         [InlineData("template/{parameter:int?}", "/template", true)]
         [InlineData("template/{parameter:int?}", "/template/qwer", false)]
-        public async Task AttributeRoute_WithOptionalInlineConstraint(
-            string template, string request, bool expectedResult)
+        public async Task TreeRouter_WithOptionalInlineConstraint(
+            string template,
+            string request,
+            bool expectedResult)
```
Comment (dougbu) [inline test/Microsoft.AspNet.Routing.Tests/Tree/TreeRouterTest.cs]: Doesn't this just fit on one line?

**Comment by dougbu** [comment]
:shipit:

File: test/Microsoft.AspNet.Routing.Tests/Tree/TreeRouterTest.cs (hunk 2 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Routing.Tests/Tree/TreeRouterTest.cs]: It didn't before the renames. I'm just going to leave it, it's not wrong.

**Pull request merged**